packages/broker/src/ConfigWizard.ts

```diff
@@ -82,200 +84,164 @@ const DefaultConfig: Config = {
-            choices: [
-                {name: 'Websocket'},
-                {name:'MQTT'},
-                {name:'HttpPublish'}
-            ]
-        })
+const pluginTemplates = {
+    websocket: { port: DEFAULT_WS_PORT },
+    mqtt: { port: DEFAULT_MQTT_PORT },
+    publishHttp: { port: DEFAULT_HTTP_PORT }
+}
 
-        const selectedPlugins = []
-        for (let i = 0; i < plugins.length; i++) {
-            selectedPlugins.push(plugins[i])
-            if (plugins[i] === 'Websocket') {
-                const wsPort = await this.promptNumberWithDefault({
-                    type: 'input',
-                    name: 'wsPort',
-                    message: `Select a port for the Websocket Plugin [Enter for default: ${DEFAULT_WS_PORT}]`,
-                }, DEFAULT_WS_PORT)
-                this.config.plugins['websocket'] = {
-                    port: wsPort,
-                    payloadMetadata: false,
-                    sslCertificate: null
-                }
-            }
+const pluginSelectorPrompt = {
+    type: 'checkbox',
+    name:'selectPlugins',
+    message: 'Select the plugins to enable',
+    choices: Object.keys(pluginTemplates)
+}
 
-            if (plugins[i] === 'MQTT') {
-                const mqttPort = await this.promptNumberWithDefault({
-                    type: 'input',
-                    name: 'mqttPort',
-                    message: `Select a port for the MQTT Plugin [Enter for default: ${DEFAULT_MQTT_PORT}]`,
-                }, DEFAULT_MQTT_PORT)
-                this.config.plugins['mqtt'] = {
-                    port: mqttPort,
-                    payloadMetadata: false
-                }
+const pluginPrompts = Object.values(pluginTemplates).map((plugin, i) => {
+    const pluginName = Object.keys(pluginTemplates)[i]
+    return {
+        type: 'input',
+        name: `${pluginName}Port`,
+        message: `Select a port for the ${pluginName} Plugin [Enter for default: ${plugin.port}]`,
+        when: (answers: inquirer.Answers) => {
+            return answers.selectPlugins.includes(pluginName)
+        },
+        validate: (input: string, answers: inquirer.Answers): string | boolean => {
+            const portNumber = parseInt(input || answers[`${pluginName}Port`])
+            if (!Number.isInteger(portNumber)) {
+                return `Non-numeric value provided`
             }
-
-            if (plugins[i] === 'HttpPublish') {
-                this.config.plugins['legacyPublishHttp'] = {}
+   
+            if (portNumber < MIN_PORT_VALUE || portNumber > MAX_PORT_VALUE) {
+                return `Out of range port ${portNumber} provided (valid range ${MIN_PORT_VALUE}-${MAX_PORT_VALUE})`
             }
+   
+            return true
+        },
+        default: plugin.port
+    }
+})
+
+prompts = prompts.concat(pluginSelectorPrompt).concat(pluginPrompts)
+
+export const getConfigFromAnswers = (answers: any): Config => {
+    const config = DEFAULT_CONFIG
```

Comment:
Should we clone the DEFAULT_CONFIGS to the config variable? Now we are modifying the exported constant value (which is used e.g. in tests)

Reply:
Something like `const config = Object.assign({}, DEFAULT_CONFIG)`?

Reply:
Or https://lodash.com/docs/4.17.15#cloneDeep, if you change e.g. the contents of config.plugins

Reply:
is it worth installing a whole new package just for this functionality?

Reply:
True, maybe easiest solution is to clone both config and plugins:
```
const config = { ... DEFAULT_CONFIG, plugins: { ... DEFAULT_CONFIG.plugins } }
```